cmd/kube-rbac-proxy/app/sanitazion.go

```diff
@@ -0,0 +1,66 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	v1 "k8s.io/api/authentication/v1"
+	"strings"
+)
+
+// SanitizingFilter implements the LogFilter interface from klog with custom functions to detect and mask tokens.
+type SanitizingFilter struct{}
+
+// Filter is the filter function for non-formatting logging functions of klog.
+func (sf *SanitizingFilter) Filter(args []interface{}) []interface{} {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
+			if strings.Contains(strValue, `"kind":"TokenReview"`) {
```

Comment:
I am curious if there are certain optimizations for this possible:
- Enable this Filter only when logging is set to a certain verbosity, which cause the tokens to be logged in the first place.
- Check for prefix `{"apiVersion": "authentication.k8s.io/v1", "kind: TokenReview"` instead of `"kind":"TokenReview"` in the whole text.
- And if we could use the json encoding error, would be more performant / sufficient (then we would pass the TokenReview object into maskTokenInLog).

Reply:
Thanks for your suggestions! I thought about each one:

- I believe it's important to always have the filter active to ensure tokens are masked in logs all time, so that we prevent any accidnetal token exposure, even if the logging level changes.
- If I'm not mistaken logs might not always start with the same prefix or have fields in the same order.
- If I understood correctly what you mean then I think it'll slow down the performance (especially if most lines aren't `TokenReview` logs). When we check for the substring first-- we skip irrelevant logs and only process the ones that matter.

I would keep the current solution, but let me know what you think :)

Reply:
Sure lets go ahead with the current solution.